app/Models/UserRecentPage.php

```
@@ -0,0 +1,143 @@
+        ];
+
+        if ($isPinned) {
+            // If pinned, add back to pinned list (preserve position by pinned_at)
+            $pinned->push($newEntry);
+            $pinned = $pinned->sortByDesc('pinned_at')->values();
+        } else {
+            // If not pinned, prepend to unpinned list
+            // Keep up to 10 unpinned for backfill when items are pinned
+            $unpinned->prepend($newEntry);
+            $unpinned = $unpinned->take(10)->values();
+        }
```

PHPMD (2.15.0)

The method recordVisit uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)